src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -239,6 +240,20 @@ module.exports = function<T, P, I, TI, C, CX>(
             currentHostContext,
             workInProgress
           );
+
+          // A component cannot be focused until it's mounted.
+          // Flag components with autoFocus for later work.
+          switch (type) {
+            case 'button':
+            case 'input':
+            case 'select':
+            case 'textarea':
+              if (newProps.autoFocus) {
```

Comment:
Yeah no worries. A convention I've been using is to tag PR with `[WIP]` and maybe mention in the description what isn't ready yet.